### PR TITLE
Issue #55

### PR DESCRIPTION
**Pull request by ghershfield**
- Added configuration attributes to allow for announcing && binding on different ports as well as allowing for environment variables to supersede the configuration.
- Fixed Issue #51 
- Found && fixed __-com.deciphernow.server.config.ipAddress.enableIpAddressResolution__ no matter the value, if defined, always true.

### CODE REVIEW

File: core/src/main/scala/com/deciphernow/server/GMFNetworkConfigurationResolver.scala [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+  /**
+    * Iterates through the interfaces for a network looking for a valid interface to register with ZK.
+    * @param networkInterface
+    */
+  protected [this] def findNetworkInfo(networkInterface: NetworkInterface) : Unit = {
+
+    if (networkInterface.isUp && !networkInterface.isLoopback && !networkInterface.isVirtual) {
+      val addresses = networkInterface.getInetAddresses
+
+      while (addresses.hasMoreElements && !(announceThis.trim.length > 0)) {
```
Comment (joshua-rutherford) [inline core/src/main/scala/com/deciphernow/server/GMFNetworkConfigurationResolver.scala]: This makes @jeffusan and @mikejlong60 cry.  Can we use the implicit conversions to make this a Scala Iterator.

https://stackoverflow.com/questions/5183730/how-to-convert-enumeration-to-seq-list-in-scala. 

I'm not hard over on this though.

**Comment by joshua-rutherford** [comment]
@mikejlong60 and @jeffusan, feel free to review if you want.

File: core/src/main/scala/com/deciphernow/server/GMFAnnouncer.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import com.twitter.logging.Logger
+
+/**
+  *
+  */
+object GMFAnnouncer {
+
+  val log = Logger.get(getClass)
+
+  private[this] var announcements = List.empty[Future[Announcement]]
```
[the remark this answers is not in the document]
Comment (ghershfield) [reply core/src/main/scala/com/deciphernow/server/GMFAnnouncer.scala]: Actually it is a list that is built over time in : def announce(port : Int, scheme: String) : Unit

File: core/src/main/scala/com/deciphernow/server/GMFNetworkConfigurationResolver.scala [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+           envValue(serverConfig.os.env.hostname.apply),
+           announceThis)
+    )
+  }
+
+  /**
+    *
+    */
+  protected [this] def resolveBindAdminPort : Unit = {
+    bindAdminPort = addTwitterNuance(
```
[the remark this answers is not in the document]
Comment (ghershfield) [reply core/src/main/scala/com/deciphernow/server/GMFNetworkConfigurationResolver.scala]: No idea what u mean.

File: core/src/main/scala/com/deciphernow/server/GMFAnnouncer.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshua-rutherford) [reply core/src/main/scala/com/deciphernow/server/GMFAnnouncer.scala]: Yes, I saw that. It's is assigned there, but not read anywhere.  It's like it crosses the event horizon.

File: core/src/main/scala/com/deciphernow/server/GMFNetworkConfigurationResolver.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshua-rutherford) [reply core/src/main/scala/com/deciphernow/server/GMFNetworkConfigurationResolver.scala]: Actually, you are probably right.  Didn't really see this was a single method invocation broken on separate lines.  Disregard.

File: core/src/main/scala/com/deciphernow/server/GMFNetworkConfigurationResolver.scala [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    }
+    retValue
+  }
+
+  /**
+    * Retrieve the value from the environment variable. Return empty String if None.
+    * @param env
+    * @return
+    */
+  protected [this] def envValue(env: Option[String]) = env.flatMap{value => sys.env.get(value)}.getOrElse("")
```
[the remark this answers is not in the document]
Comment (joshua-rutherford) [reply core/src/main/scala/com/deciphernow/server/GMFNetworkConfigurationResolver.scala]: So here is the idomatic way:

```
env.fold("")(sys.env.getOrElse(value, ""))
```

File: core/src/main/scala/com/deciphernow/server/GMFAnnouncer.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshua-rutherford) [reply core/src/main/scala/com/deciphernow/server/GMFAnnouncer.scala]: @ghershfield, you still have this private variable that is only ever assigned, what is it used for?

**Pull request merged**